### PR TITLE
Implementation of 'Notification' and 'Bluetooth'  permissions

### LINKED ISSUE #493: Android 11+ invalid runtime permissions handling for "ask again later" answer
### Description

Example. 
User denied access to location 2 times. After that user goes to settings and choose 'one time access'. As result permission is not requested again, but android can show dialog again.  

### Steps to Reproduce

```cs
private async Task<PermissionStatus> CommonCheckWithRequestAsync<T>()
            where T : BasePermission, new()
{
    var permissionStatus = await _permissionsService.CheckPermissionsAsync<T>().ConfigureAwait(false);
    if (permissionStatus == PermissionStatus.Granted)
    {
        return permissionStatus;
    }

    ApplyKeysMigration<T>(permissionStatus);

    var shouldRationalate = _permissionsService.ShouldShowRationale<T>();
    if (permissionStatus == PermissionStatus.Denied
        && IsPermissionDeniedEver<T>()
        && !_permissionsService.ShouldShowRationale<T>())
    {
        await OpenSettingsWithConfirmationAsync<T>().ConfigureAwait(false);
        return PermissionStatus.Denied;
    }

    var confirmationResult = await _permissionsDialogService.ConfirmPermissionAsync<T>().ConfigureAwait(false);
    if (confirmationResult)
    {
        permissionStatus = await _permissionsService.RequestPermissionsAsync<T>().ConfigureAwait(false);
        SetPermissionDenied<T>(permissionStatus == PermissionStatus.Denied);
    }

    return permissionStatus;
}
```

We store in cache that permission is ever denied, but if status is changed in settings to 'ask again later' then ` IsPermissionDeniedEver<T>()` return false, permission status is Denied and as result permission is not asked again.

### Expected Behavior

Permission should be asked

### Actual Behavior

Permission is not asked

## Possible Fix

Implement permission mechanism according to android guidelines https://developer.android.com/training/permissions/requesting#workflow_for_requesting_permissions

### Basic Information

- Platform Target Frameworks: 
  - Android: 11+

### PR DESCRIPTION
**Pull request by nkrisko**
### Description
Added check and request for custom permissions of 'Notification' and 'Bluetooth' types

### Issues Resolved

- fixes #493 

### API Changes

Added:
 - NuGet -  [System.Reactive](https://www.nuget.org/packages/System.Reactive) to :
   - Softeq.XToolkit.Permissions.iOS

 - Softeq.XToolkit.Permissions/PermissionsService.cs
 
 Removed:
 - Softeq.XToolkit.Permissions.iOS/PermissionsService.cs
 - Softeq.XToolkit.Permissions.Droid/PermissionsService.cs

### Platforms Affected

- Core (all platforms)
- iOS
- Android

### Behavioral/Visual Changes

None

### Before/After Screenshots

Not applicable

### PR Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/Softeq/XToolkit.WhiteLabel/blob/master/.github/CONTRIBUTING.md) document
- [x] My code follows the [code styles](https://github.com/Softeq/dotnet-guidelines/tree/xamarin_guidelines)
- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: samples/Playground/Playground.iOS/Softeq.XToolkit.WhiteLabel.Essentials.iOS/FullScreenImage/FullScreenImageViewController.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// WARNING
+// This file has been generated automatically by Visual Studio to
+// mirror C# types. Changes in this file made by drag-connecting
+// from the UI designer will be synchronized back to C#, but
+// more complex manual changes may not transfer correctly.
+
+
+#import <Foundation/Foundation.h>
```
Comment (wcoder) [inline samples/Playground/Playground.iOS/Softeq.XToolkit.WhiteLabel.Essentials.iOS/FullScreenImage/FullScreenImageViewController.h:8]: Are native files required? looks strange

File: Softeq.XToolkit.Permissions.iOS/Softeq.XToolkit.Permissions.iOS.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0-ios10.0</TargetFramework>
+    <TargetFramework>net6.0-ios14.0</TargetFramework>
```
Comment (wcoder) [inline Softeq.XToolkit.Permissions.iOS/Softeq.XToolkit.Permissions.iOS.csproj:4]: Looks like ios13 will be enough,
only bluetooth requires ios13+

**Review by wcoder — CHANGES_REQUESTED** [review]

**Comment by wcoder** [comment]
From CI logs:
> ##[error]The job running on agent Hosted Agent ran longer than the maximum time of 60 minutes. For more information, see https://go.microsoft.com/fwlink/?linkid=2077134

File: samples/Playground/Playground.iOS/Softeq.XToolkit.WhiteLabel.Essentials.iOS/FullScreenImage/FullScreenImageViewController.h (hunk 1 above)
Comment (nkrisko) [reply samples/Playground/Playground.iOS/Softeq.XToolkit.WhiteLabel.Essentials.iOS/FullScreenImage/FullScreenImageViewController.h:8]: removed

File: Softeq.XToolkit.Permissions.Droid/Permissions/Notifications.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+using Android.OS;
+#endif
+using EssentialsPermissions = Microsoft.Maui.ApplicationModel.Permissions;
+
+namespace Softeq.XToolkit.Permissions.Droid.Permissions
+{
+    public class Notifications : EssentialsPermissions.BasePlatformPermission
+    {
+        public override (string, bool)[] RequiredPermissions =>
+#if __ANDROID_33__
```
[the remark this answers is not in the document]
Comment (nkrisko) [reply Softeq.XToolkit.Permissions.Droid/Permissions/Notifications.cs:15]: changed

File: Softeq.XToolkit.Permissions.Droid/PermissionsManager.cs [hunk 4]
```diff
@@ -101,21 +113,22 @@ private async Task<PermissionStatus> CommonCheckWithRequestAsync<T>()
...
+            RemoveOldKeys<T>();
 
-            if (permissionStatus == PermissionStatus.Denied
-                && IsPermissionDeniedEver<T>()
-                && !_permissionsService.ShouldShowRationale<T>())
-            {
-                await OpenSettingsWithConfirmationAsync<T>().ConfigureAwait(false);
-                return PermissionStatus.Denied;
-            }
+            var timer = new Stopwatch();
```
[the remark this answers is not in the document]
Comment (nkrisko) [reply Softeq.XToolkit.Permissions.Droid/PermissionsManager.cs:118]: added

File: Softeq.XToolkit.Permissions/Permissions/Notifications.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+namespace Softeq.XToolkit.Permissions
```
[the remark this answers is not in the document]
Comment (nkrisko) [reply Softeq.XToolkit.Permissions/Permissions/Notifications.cs:4]: fixed

File: Softeq.XToolkit.Permissions/Permissions/Bluetooth.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+namespace Softeq.XToolkit.Permissions
```
[the remark this answers is not in the document]
Comment (nkrisko) [reply Softeq.XToolkit.Permissions/Permissions/Bluetooth.cs:4]: fixed

File: Softeq.XToolkit.Permissions.Droid/SdkVersion.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using Android.OS;
+using Microsoft.Maui.ApplicationModel;
+
+namespace Softeq.XToolkit.Permissions.Droid
+{
+    public static class SdkVersion
```
[the remark this answers is not in the document]
Comment (nkrisko) [reply Softeq.XToolkit.Permissions.Droid/SdkVersion.cs:9]: made 'internal'

File: Softeq.XToolkit.Permissions.iOS/Permissions/Bluetooth.cs [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            if (status == EssentialsPermissionStatus.Granted)
+            {
+                return status;
+            }
+
+            if (CBManager.Authorization == CBManagerAuthorization.NotDetermined)
+            {
+                var centralMamangerDelegate = new CentralManagerDelegate();
+                await centralMamangerDelegate.StatusObserverable
+                    .Select(x => x != CBManagerState.Unknown)
```
[the remark this answers is not in the document]
Comment (nkrisko) [reply Softeq.XToolkit.Permissions.iOS/Permissions/Bluetooth.cs:43]: improved

File: Softeq.XToolkit.Permissions.iOS/Permissions/Bluetooth.cs [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+            var status = await CheckStatusAsync();
+
+            if (status == EssentialsPermissionStatus.Granted)
+            {
+                return status;
+            }
+
+            if (CBManager.Authorization == CBManagerAuthorization.NotDetermined)
+            {
+                var centralMamangerDelegate = new CentralManagerDelegate();
```
[the remark this answers is not in the document]
Comment (nkrisko) [reply Softeq.XToolkit.Permissions.iOS/Permissions/Bluetooth.cs:41]: fixed

File: Softeq.XToolkit.Permissions.iOS/Permissions/Bluetooth.cs [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+            private readonly CBCentralManager _manager;
+            private readonly Subject<CBManagerState> _subject
+                = new Subject<CBManagerState>();
+
+            public CentralManagerDelegate()
+            {
+                _manager = new CBCentralManager(this, null);
+            }
+
+            public IObservable<CBManagerState> StatusObserverable => _subject;
```
[the remark this answers is not in the document]
Comment (nkrisko) [reply Softeq.XToolkit.Permissions.iOS/Permissions/Bluetooth.cs:74]: not actual, removed

File: Softeq.XToolkit.Permissions.iOS/Softeq.XToolkit.Permissions.iOS.csproj (hunk 2 above)
Comment (nkrisko) [reply Softeq.XToolkit.Permissions.iOS/Softeq.XToolkit.Permissions.iOS.csproj:4]: changed

**Review by wcoder — APPROVED** [review]

**Pull request merged**